src/system/PageLockModule/Api/LockingApi.php

```diff
@@ -0,0 +1,262 @@
+                if (strlen($lockedBy) > 0) {
+                    $lockedBy .= ', ';
+                }
+                $lockedBy .= $lock['title'] . " ($lock[ipno]) " . $lock['cdate']->format('Y-m-d H:m:s');
+            }
+
+            return ['hasLock' => false, 'lockedBy' => $lockedBy];
+        }
+
+        // Look for existing lock
+        $count = $this->getRepository()->getLockAmount($lockName, $theSessionId);
+
+        $now = time();
+        $expireDate = $now + self::PAGELOCKLIFETIME;
+
+        if ($count > 0) {
+            // update the existing lock with a new expiry date
+            $this->getRepository()->updateExpireDate($lockName, $theSessionId, $expireDate);
+        } else {
+            // create the new object
+            $newLock = new PageLockEntity();
+            $newLock->setName($lockName);
+            $newLock->setCdate(new \DateTime(\DateUtil::getDatetime($now)));
+            $newLock->setEdate(new \DateTime(\DateUtil::getDatetime($expireDate)));
+            $newLock->setSession($theSessionId);
+            $newLock->setTitle($lockedByTitle);
+            $newLock->setIpno($lockedByIPNo);
+            // write this back to the db
+            $this->entityManager->persist($newLock);
+        }
+        $this->entityManager->flush();
+
+        $this->releaseAccess();
+
+        return ['hasLock' => true];
+    }
+
+    /**
+     * Get all the locks for a given page
+     *
+     * @param string $lockName  The name of the page to return locks for
+     * @param string $sessionId The ID of the session owning the lock (optional) (default: current session ID)
+     *
+     * @return array array of locks for $lockName
+     */
+    public function getLocks($lockName, $sessionId = '')
+    {
+        $theSessionId = $sessionId != '' ? $sessionId : session_id();
+
+        $this->requireAccess();
+
+        // remove expired locks
+        $this->getRepository()->deleteExpiredLocks();
+
+        // get remaining active locks
+        $locks = $this->getRepository()->getActiveLocks($lockName, $theSessionId);
+
+        $this->releaseAccess();
+
+        return $locks;
+    }
+
+    /**
+     * Releases a lock on a page
+     *
+     * @param string $lockName  The name of the lock to be released
+     * @param string $sessionId The ID of the session owning the lock (optional) (default: current session ID)
+     *
+     * @return bool true
+     */
+    public function releaseLock($lockName, $sessionId = '')
+    {
+        $theSessionId = $sessionId != '' ? $sessionId : session_id();
+
+        $this->requireAccess();
+
+        $this->getRepository()->deleteByLockName($lockName, $theSessionId);
+
+        $this->releaseAccess();
+
+        return true;
+    }
+
+    /**
+     * Returns repository for the page lock entities.
+     *
+     * @return Doctrine\ORM\EntityRepository
+     */
+    private function getRepository()
+    {
+        return $this->getDoctrine()->getManager()
```

Comment:
example:

https://github.com/zikula/core/blob/1.4/src/system/BlocksModule/Resources/config/services.xml#L9

and

https://github.com/zikula/core/blob/1.4/src/system/BlocksModule/Resources/config/services.xml#L28-L31

and finally

https://github.com/zikula/core/blob/1.4/src/system/BlocksModule/Resources/config/services.xml#L38